### PR TITLE
Backport for release of azure-cosmos_4.45.3-hotfix

### PR DESCRIPTION
**Pull request by jeet1995**
Changes summary:
- Following PRs have been cherry-picked:
  - #39077
  - #37015

This along with removing `System.exit()` calls from `ImplementationBridgeHelpers`.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/util/CosmosPagedFlux.java [hunk 1]
```diff
@@ -145,15 +145,109 @@ private CosmosPagedFluxOptions createCosmosPagedFluxOptions() {
...
+                    }
+                }
+            });
+
+        return Flux
+            .deferContextual(reactorCtx -> result
+                .doOnCancel(() -> {
+                    Context traceCtx = DiagnosticsProvider.getContextFromReactorOrNull(reactorCtx);
+                    synchronized (lockHolder) {
+                        tracerProvider.endSpan(traceCtx);
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/util/CosmosPagedFlux.java:238]: Can you please do me one favor and triple check that exception (non-errors) leaking from traceProvider.endSPan don't cause the Flux to hang in any of the stages where endSpan is called? Annie might have validated it already - but that would be useful to check at least manually.

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,14 @@
 ## Release History
 
+### 4.46.0-beta.1 (Unreleased)
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:3]: will also need to update the azure-cosmos sdk version in pom.xml to 4.46.0-beta.1, else the pipeline will fail

and then you will need to also update all the versions mentioned in 
`{x-version-update;com.azure:azure-cosmos;dependency} `
`{x-version-update;com.azure:azure-cosmos;current}`

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos/CHANGELOG.md (hunk 2 above)
Comment (jeet1995) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md:3]: I can follow up on this in the release branch / PR.

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/implementation/ImplementationBridgeHelpersTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import org.testng.annotations.Test;
+
+import java.lang.reflect.Field;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+public class ImplementationBridgeHelpersTest {
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/implementation/ImplementationBridgeHelpersTest.java:21]: Thanks - I love this test!

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

File: sdk/cosmos/azure-cosmos/CHANGELOG.md (hunk 2 above)
Comment (jeet1995) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md]: Ran the prepare release script to update required files to specify 4.45.3-hotfix.

**Comment by jeet1995** [comment]
Observations from the live tests pipeline:
- `clientTelemetryWithStageJunoEndpoint` is failing - this has been fixed in a later PR - unrelated to this PR.
- `conflictCustomLWW` - known flaky test.
- `readMany` - had a known bug `4.45.x` versions fixed in `4.47.0`.
- `recordNewAddressesAfterSplit` - passes locally.

Speculation related and client-telemetry related tests have been removed in future versions.

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

**Pull request merged**